pet/modules/TransactionHistory/TransactionHistoryPage.tsx

```diff
@@ -0,0 +1,132 @@
+'use client';
+
+import React from 'react'
+import { useEffect, useState } from 'react';
+import CenterLoader from '@components/CenterLoader';
+import TransactionFilter from '@components/TransactionFilter';
+import Transaction from '@components/Transaction/index';
+import { Typography, Grid, createTheme, ThemeProvider, Box } from '@mui/material';
+import useGetTransactionHistory from '@services/api/v1/user/useGetTransactionHistory';
+import Image from 'next/image';
+import dogSleep from '@public/image/dogSleep.png';
+import { set } from 'react-hook-form';
+
+import { Fira_Sans_Condensed } from 'next/font/google';
+
+const fira_sans_condensed = Fira_Sans_Condensed({ weight: ['600'], subsets: ['latin'] });
+
+const theme = createTheme({
+    typography: {
+        fontFamily: fira_sans_condensed.style.fontFamily,
+    },
+});
+
+function TransactionHistoryPage() {
+    const [data, setData] = useState([]);
+    const [role, setRole] = useState(1);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            // Mock API call
+            const response : any = await useGetTransactionHistory();
+            const formattedData = response.data.map((transaction: any) => {
+                if (transaction.payment_method.toLowerCase() === 'paypal') {
+                    transaction.payment_method = 'PAYPAL';
+                } else if (transaction.payment_method.toLowerCase() === 'creditcard') {
+                    transaction.payment_method = 'CREDIT CARD';
+                }
+                if (role === 1) {
+                    return {
+                        ...transaction,
+                        price: '+' + transaction.price.toLocaleString() + ' THB'
+                    }
+                } else if (role === 2) {
+                    return {
+                        ...transaction,
+                        price: '-' + transaction.price.toLocaleString() + ' THB'
+                    }
+                }
+            });
+            setData(formattedData);
+            setRole(response.role);
+            setLoading(response.loading);
+            setError(response.error);
+        };
+        fetchData();
+    }, [data]);
+
+    if (loading) return <CenterLoader />;
+    // Never have error ;)
+    // if (error) return <Error />;
```

Comment:
Oh shit, there can be some errors whether from backend or database.
`const toastUI = useToastUI()` and do .error(msg) or .success(msg) if unable or able to get the data